### PR TITLE
Fix: Show updates during analysis

### LINKED ISSUE #1667: [extension-browser] Show a progress bar during analysis
# 🚀 Feature request

## Description

Based on feedback from @ststimac.

Display a progress bar (possibly an indeterminate one) after clicking "Analyze" to clearly indicate that the scan is in progress.

## What scenarios will this solve?

The current UX doesn't clearly communicate to users that webhint is processing the page. They may mistakenly think the tool is stuck/hung.

### PR DESCRIPTION
**Pull request by antross**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- ~Added/Updated related documentation.~
- ~Added/Updated related tests.~

## Short description of the change(s)

Fix #1667

Add a "working" graphic and a periodically updating status
message to indicate progress during analysis.

Did not add an explicit progress bar as hints run in parallel and
we don't have a clear indicator of how long a scan will take.

Also clean up some dead code and variable names.

### CODE REVIEW

**Comment by antross** [comment]
Bundle for testing:
* [extension-browser-fix-1667.zip](https://github.com/webhintio/hint/files/2770869/extension-browser-fix-1667.zip)

**Comment by antross** [comment]
> Maybe we can force a message to be visible for at least 5 seconds or so?

Are you thinking a minimum of 5 seconds after first switching to the "Analyzing" view? Or a minimum of 5 seconds after we switched to the most recent message? (So even if we'd already been up for 20 seconds, if we just switched messages 1 second ago we'd wait 4 more seconds)

**Comment by molant** [comment]
This:

> So even if we'd already been up for 20 seconds, if we just switched messages 1 second ago we'd wait 4 more seconds

**Comment by antross** [comment]
Updated to show messages for a minimum of 5s before advancing. Still allows the user to cancel immediately at any time.

File: packages/extension-browser/src/devtools/panel/utils/inspire.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    `52% of resources aren't cacheable`,
+    `The median site size is 1,731KB`,
+    `1MB of JavaScript takes more time to process than 1MB of a JPEG file`
+];
+
+let remaining = [...messages];
+
+const inspire = (): string => {
+    if (!remaining.length) {
+        remaining = [...messages];
```
Comment (molant) [inline packages/extension-browser/src/devtools/panel/utils/inspire.ts]: Nice catch

Comment (antross) [reply packages/extension-browser/src/devtools/panel/utils/inspire.ts]: At least the earlier version didn't run out of messages as it always selected a random index without mutating the array. It just suffered from the possibility of repeating the same message before getting through the list, potentially even back-to-back.

**Comment by antross** [comment]
@sarvaje Mind taking a look at this if you have time? Thanks!

**Comment by sarvaje** [comment]
> @sarvaje Mind taking a look at this if you have time? Thanks!

Done!

**Pull request merged**